lib/chef/resource/chocolatey_installer.rb

```diff
@@ -0,0 +1,198 @@
+
+      property :proxy_password, String,
+        description: "The password to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_user are set"
+
+      load_current_value do
+        current_state = is_choco_installed?
+        current_value_does_not_exist! if current_state == false
+        current_state
+      end
+
+      def is_choco_installed?
+        ::File.exist?("#{ENV["ALLUSERSPROFILE"]}\\chocolatey\\bin\\choco.exe")
+      end
+
+      def get_choco_version
+        powershell_exec("choco --version").result
+      end
+
+      def existing_version
+        Gem::Version.new(get_choco_version)
+      end
+
+      def define_resource_requirements
+        [ new_resource.proxy_user, new_resource.proxy_password ].each do
+          requirements.assert(:install, :upgrade).each do |a|
+            a.assertion do
+              (!new_resource.proxy_user.nil? && new_resource.proxy_password.nil?) || (new_resource.proxy_user.nil? && !new_resource.proxy_password.nil?)
+            end
+            a.failure_message(Chef::Exceptions::ValidationFailed, "You must specify both a proxy_user and a proxy_password")
+            a.whyrun("Assuming that if you have configured a 'proxy_user' you must also supply a 'proxy_password'")
+          end
+        end
+      end
+
+      action :install, description: "Installs Chocolatey package manager" do
+        unless new_resource.download_url.nil?
+          powershell_exec("Set-Item -path env:ChocolateyDownloadUrl -Value #{new_resource.download_url}")
+        end
+
+        unless new_resource.chocolatey_version.nil?
+          powershell_exec("Set-Item -path env:ChocolateyVersion -Value #{new_resource.chocolatey_version}")
+        end
+
+        if new_resource.use_native_unzip == true
+          powershell_exec("Set-Item -path env:ChocolateyUseWindowsCompression -Value true")
+        end
+
+        if new_resource.ignore_proxy == true
+          powershell_exec("Set-Item -path env:ChocolateyIgnoreProxy -Value true")
+        end
+
+        unless new_resource.proxy_url.nil?
+          powershell_exec("Set-Item -path env:ChocolateyProxyLocation -Value #{new_resource.proxy_url}")
+        end
+
+        unless !new_resource.proxy_user.nil? && !new_resource.proxy_password.nil?
+          powershell_exec("Set-Item -path env:ChocolateyProxyUser -Value #{new_resource.proxy_user}; Set-Item -path env:ChocolateyProxyPassword -Value #{new_resource.proxy_password}")
+        end
+
+        converge_if_changed do
+          powershell_exec("iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))").error!
+        end
+      end
+
+      action :upgrade, description: "Upgrades the Chocolatey package manager" do
+        if !new_resource.chocolatey_version.nil?
```

Comment:
Switch this around

If version
  ...
Else
...
End

(Sorry, typing on phone)